src/frontend/sharedComponents/GpsPill.tsx

```diff
@@ -0,0 +1,106 @@
+import * as React from 'react';
+import {View, StyleSheet} from 'react-native';
+import {defineMessages, useIntl} from 'react-intl';
+import {useIsFocused} from '@react-navigation/native';
+import {TouchableOpacity} from 'react-native-gesture-handler';
+import {useForegroundPermissions} from 'expo-location';
+
+import {BLACK, WHITE} from '../lib/styles';
+import {useLocationProviderStatus} from '../hooks/useLocationProviderStatus';
+import {useLocation} from '../hooks/useLocation';
+import {getLocationStatus} from '../lib/utils';
+import type {LocationStatus} from '../lib/utils';
+import {Text} from './Text';
+import {GpsIcon} from './icons';
+
+const m = defineMessages({
+  noGps: {
+    id: 'sharedComponents.GpsPill.noGps',
+    defaultMessage: 'No GPS',
+  },
+  searching: {
+    id: 'sharedComponents.GpsPill.searching',
+    defaultMessage: 'Searching…',
+  },
+});
+
+type Props = {
+  onPress?: () => void;
+  precision?: number;
+  variant: LocationStatus;
+};
+
+const MemoizedGpsPill = React.memo<Props>(
+  ({onPress, variant, precision}: Props) => {
+    const isFocused = useIsFocused();
+    const {formatMessage: t} = useIntl();
+    let text: string;
+    if (variant === 'error') text = t(m.noGps);
+    else if (variant === 'searching' || typeof precision === 'undefined')
+      text = t(m.searching);
+    else text = `± ${precision} m`;
+    return (
+      <TouchableOpacity onPress={onPress || undefined} testID="gpsPillButton">
+        <View
+          style={[
+            styles.container,
+            variant === 'error' ? styles.error : undefined,
+          ]}>
+          <View style={styles.icon}>
+            {isFocused && <GpsIcon variant={variant} />}
+          </View>
+          <Text style={styles.text} numberOfLines={1}>
+            {text}
+          </Text>
+        </View>
+      </TouchableOpacity>
+    );
+  },
+);
+
+export const GpsPill = ({onPress}: Pick<Props, 'onPress'>) => {
+  const locationState = useLocation({maxDistanceInterval: 15});
```

Comment:
not sure what the number here should be. using the one defined for the MapScreen

Reply:
I think this should have a `maxDistanceInterval` of 0. Setting it to 15 makes sense for the map because we do not want the map to be constantly re-rendering. With the accuracy, we do want the user to know that the observation that they are saving has the most up to date accuracy. So anytime the accuracy changes, we want that to be reflected in the UI

Reply:
This is also causing the "searching" text to appear when the user has not moved 15 meters due to the `STALE_TIMEOUT`

Reply:
addressed via 2869abd8f2f8b7b412c04a547dc13ec475d73d99